tracks/elm/exercises/leap/mentoring.md

```diff
@@ -0,0 +1,90 @@
+This is a good exercise to discuss the describability of logic, as most students solve the problem using a conditional statement that is hard to describe in language, and there are other options.
+
+This exercise introduces `Basics.remainderBy`, the `let` / `in` syntax and partial application / currying.
+
+### Reasonable solutions
+
+The easiest / most common solution probably involves partially applying `Basics.remainderBy` and using a slightly complicated if statement:
+
+```elm
+isLeapYear : Int -> Bool
+isLeapYear year =
+    let
+        divisibleBy number = 
+            Basics.remainderBy number year == 0         
+    in
+        (
+            divisibleBy 4 && 
+            not (divisibleBy 100)
+        )
+        || divisibleBy 400
+```
+
+The complicated if statement is a bit of a problem, as it is hard to describe in language, and hence talking about the code with others is difficult.
```

Comment:
What complicated if-statement?

Do you mean the `&&`/`||`/`not`-based solution above?

Reply:
I do mean the `divisibleBy 4 && (not (divisibleBy 100) || divisibleBy 400)` statement.

One of my main points is that this is very hard to describe in words (to a product owner or a business analyst or another programmer or whatever).